oarepo_ui/theme/assets/semantic-ui/js/oarepo_ui/forms/components/EDTFDatePickerField/EDTFDateRangePickerField.jsx

```diff
@@ -0,0 +1,253 @@
+  startDateInputPlaceholder,
+  endDateInputPlaceholder,
+  singleDateInputPlaceholder,
+  ...datePickerProps
+}) => {
+  // TODO: The datepickers shall recieve needed locales from form config (set in Invenio.cfg)
+  const { setFieldValue } = useFormikContext();
+  const [field] = useField(fieldPath);
+  const initialEdtfDateFormat = getInitialEdtfDateFormat(field?.value);
+  const [dateEdtfFormat, setDateEdtfFormat] = useState(initialEdtfDateFormat);
+  const [showSingleDatePicker, setShowSingleDatePicker] = useState(false);
+  let dates;
+  if (field?.value) {
+    dates = field.value.split("/").map((date) => deserializeDate(date));
+  } else {
+    dates = [null, null];
+  }
+
+  const startDate = dates[0];
+  const endDate = dates[1];
+
+  const handleChange = (dates) => {
+    const serializedDates = dates.map((date) =>
+      serializeDate(date, dateEdtfFormat)
+    );
+    if (allEmptyStrings(serializedDates)) {
+      setFieldValue(fieldPath, "");
+    } else {
+      setFieldValue(fieldPath, serializedDates.join("/"));
+    }
+  };
+
+  const handleSingleDateChange = (date) => {
+    handleChange([date, date]);
+  };
+
+  const handleClearStartDate = () => {
+    dates[0] = null;
+    handleChange(dates);
+  };
+  const handleClearEndDate = () => {
+    dates[1] = null;
+    handleChange(dates);
+  };
+
+  const handleClearSingleDate = () => {
+    handleChange([null, null]);
+  };
+  // handle situation if someone selected just one date, when switching to the single input
+  // to fill it with one of the selected values
+  const handleSingleDatePickerSelection = () => {
+    if (!dates[0] && dates[1]) {
+      const newDates = [dates[1], dates[1]].map((date) =>
+        serializeDate(date, dateEdtfFormat)
+      );
+      setFieldValue(fieldPath, newDates.join("/"));
+    } else if (!dates[1] && dates[0]) {
+      const newDates = [dates[0], dates[0]].map((date) =>
+        serializeDate(date, dateEdtfFormat)
+      );
+      setFieldValue(fieldPath, newDates.join("/"));
+    }
+    setShowSingleDatePicker(true);
+  };
+  useLoadLocaleObjects();
+  return (
+    <React.Fragment>
+      <Form.Field className="ui datepicker field mb-0" required={required}>
+        <FieldLabel htmlFor={fieldPath} icon={icon} label={label} />
+        <Form.Field className="mb-0">
+          <Radio
+            label={i18next.t("Choose start and end date.")}
+            name="startAndEnd"
+            checked={!showSingleDatePicker}
+            onChange={() => setShowSingleDatePicker(false)}
+            className="rel-mr-1"
+          />
+          <Radio
+            label={i18next.t("Choose one date as start and end date.")}
+            name="oneDate"
+            checked={showSingleDatePicker}
+            onChange={() => handleSingleDatePickerSelection()}
+            required={false}
+          />
+        </Form.Field>
+        {showSingleDatePicker ? (
+          <div>
+            <DatePicker
+              locale={i18next.language}
```

Comment:
Ive seen this as well, but not really because the two datepickers are a combination ( the ones that make the range picker) and props do look similar, but are not actually